src/body.rs

```diff
@@ -175,6 +175,9 @@ pub trait LineMetaMut: LineMeta {
     fn set_labels(&mut self, labels: KeyValueMap) -> Result<(), LineMetaError>;
     fn set_level(&mut self, level: String) -> Result<(), LineMetaError>;
     fn set_meta(&mut self, meta: Value) -> Result<(), LineMetaError>;
+    fn get_line(&mut self) -> (Option<&str>, Option<&[u8]>);
```

Comment:
Could these methods for dealing with the lines be split out into a separate trait?

The idea behind having the Meta traits was that "lines" were just types that could serialize to the buffer, they don't actually need to be backed by data in memory. 

Ie types where the line wasn't actually available to read or write to, it's opaque right up until it's serialized.